src/server-extension/resolvers/VideosResolver/index.ts

```diff
@@ -134,29 +134,26 @@ export class VideosResolver {
 
     let connectionQuerySql: string
 
+    connectionQuerySql = extendClause(
+      connectionQuery.sql,
+      'WHERE',
+      `"video"."id" IN (${ids.map((id) => `'${id}'`).join(', ')})`,
+      'AND'
+    )
+
     const hasPeriodDaysArgAndIsOrderedByViews =
       args.periodDays &&
       (args.orderBy.find((orderByArg) => orderByArg === 'viewsNum_DESC') ||
         args.orderBy.find((orderByArg) => orderByArg === 'viewsNum_ASC'))
 
     if (hasPeriodDaysArgAndIsOrderedByViews) {
-      connectionQuerySql = extendClause(
-        connectionQuery.sql,
-        'FROM',
-        `JOIN unnest('{${ids
-          .map((id) => `"${id}"`)
-          .join(', ')}}'::text[]) WITH ORDINALITY t(id, ord) USING ("id")`,
-        ''
-      )
-
-      connectionQuerySql = connectionQuerySql.replace('"video"."views_num" DESC', 't.ord')
-      connectionQuerySql = connectionQuerySql.replace('"video"."views_num" ASC', 't.ord DESC')
-    } else {
-      connectionQuerySql = extendClause(
-        connectionQuery.sql,
-        'WHERE',
-        `"video"."id" IN (${ids.map((id) => `'${id}'`).join(', ')})`,
-        'AND'
+      connectionQuerySql = overrideClause(
```

Comment:
When you're using `overrideClause` you're replacing the entire `ORDER BY` clause of the query, which means that if multiple `orderBy` values are provided and one of them is `viewsNum_DESC` (for example: `orderBy: [createdAt_DESC, viewsNum_DESC],`), all other values will be ignored.

So perhaps the previous approach where you used:
```typescript
connectionQuerySql.replace('"video"."views_num" DESC', /* ... */)
```
What better in the sense that it was just replacing the specific part of the `ORDER BY` caluse.
Of course in very complex queries `"video"."views_num" DESC` may appear multiple times (for example, in some subqueries also), but I'm not sure if it's worth worrying about, as it would be an extremely rare case.